lib/commons/matches/from-definition.js

```diff
@@ -0,0 +1,32 @@
+/* global matches */
+const matchers = [
+  'nodeName',
+  'attributes',
+  'properties',
+  'condition'
+]
+
+matches.fromDefinition = function matchFromDefinition (node, definition) {
+  node = node.actualNode || node;
+	if (Array.isArray(definition)) {
+		return definition.some(definition => matches(node, definition));
+  }
+	if (typeof definition === 'string') {
+		return axe.utils.matchesSelector(node, definition);
+  }
+
+  return Object.keys(definition)
+  .every(matcherName => {
+    if (!matchers.includes(matcherName)) {
+      console.log(definition)
+      throw new Error(`Unknown matcher type "${matcherName}"`)
+    }
+    const matchMethod = matches[matcherName]
+    const matcher = definition[matcherName]
+    return matchMethod(node, matcher)
+  })
+}
+
+matches.condition = function (node, condition) {
```

Comment:
There are several "condition" things in the aria/index.js file. I didn't put it in its own file because I'm lazy, and it's a single line function. Didn't seem worth the effort. Please let me know if you think this needs to be done.

Reply:
Just for separation of concerns, I would put this in a single file.

Reply:
It definitely needs some comments about usage. Similar to `matches.attributes`, it's difficult to follow how the lookupTable functions with this matches object...